Engine/DataFeeds/DataManager.cs

```diff
@@ -67,6 +119,98 @@ public class DataManager : IAlgorithmSubscriptionManager, IDataFeedSubscriptionM
+                }
+            }
+        }
+
+        /// <summary>
+        /// Adds a new <see cref="Subscription"/> to provide data for the specified security.
+        /// </summary>
+        /// <param name="request">Defines the <see cref="SubscriptionRequest"/> to be added</param>
+        /// <returns>True if the subscription was created and added successfully, false otherwise</returns>
+        public bool AddSubscription(SubscriptionRequest request)
+        {
+            if (DataFeedSubscriptions.Contains(request.Configuration))
+            {
+                // duplicate subscription request
+                return false;
+            }
+
+            var subscription = _dataFeed.CreateSubscription(request);
+
+            if (subscription == null)
+            {
+                Log.Trace($"DataManager.AddSubscription(): Unable to add subscription for: {request.Configuration}");
+                // subscription will be null when there's no tradeable dates for the security between the requested times, so
+                // don't even try to load the data
+                return false;
+            }
+
+            Log.Debug($"DataManager.AddSubscription(): Added {request.Configuration}." +
+                $" Start: {request.StartTimeUtc}. End: {request.EndTimeUtc}");
+            DataFeedSubscriptions.TryAdd(subscription);
+            return true;
+        }
+
+        /// <summary>
+        /// Removes the <see cref="Subscription"/>, if it exists
+        /// </summary>
+        /// <param name="configuration">The <see cref="SubscriptionDataConfig"/> of the subscription to remove</param>
+        /// <returns>True if the subscription was successfully removed, false otherwise</returns>
+        public bool RemoveSubscription(SubscriptionDataConfig configuration)
+        {
+            // remove the subscription from our collection, if it exists
+            Subscription subscription;
+
+            if (DataFeedSubscriptions.TryGetValue(configuration, out subscription))
+            {
+                // don't remove universe subscriptions immediately, instead mark them as disposed
+                // so we can turn the crank one more time to ensure we emit security changes properly
+                if (subscription.IsUniverseSelectionSubscription && subscription.Universe.DisposeRequested)
+                {
+                    // subscription syncer will dispose the universe AFTER we've run selection a final time
+                    // and then will invoke SubscriptionFinished which will remove the universe subscription
+                    return false;
+                }
+
+                if (!DataFeedSubscriptions.TryRemove(configuration, out subscription))
+                {
+                    Log.Error($"DataManager.RemoveSubscription(): Unable to remove {configuration}");
+                    return false;
+                }
+
+                _dataFeed.RemoveSubscription(subscription);
```

Comment:
In `AddSubscription` we only interact with the `DataFeedSubscriptions` but here we're removing directly from `IDataFeed` -- why the inconsistency? Also, should be handling the return value from `RemoveSubscription` in the event it wasn't removed from the DF?

Reply:
Both while adding and removing we interact with both `DataFeedSubscriptions` and `IDataFeed`. While adding we call `_dataFeed.CreateSubscription()` and `DataFeedSubscriptions.TryAdd()`. The `DF` does not handle the Subscription at this point anymore. Its used as a cleanup process, which only is used in the `LiveTradingDataFeed`

Reply:
very awesome, a future improvement could be implementing the live-specific clean up as a dispose method and that _should_ end up by removing `RemoveSubscription` from `IDataFeed` which _should_ completely de-couple the data feeds from subscription manage 🎆